### PR TITLE
WebTorrent: Fix stats remain stuck after download finishes

### PR DESCRIPTION
**Pull request by feross**
Fixes: https://github.com/brave/brave-browser/issues/5352

- Always update stats at least once every 5 seconds, even when no torrent events fire
- remove stray console logs
- Listen for 'wire' (new peer) events
- Listen for 'done' event

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

   1. Visit https://webtorrent.io/free-torrents
   2. Click Big Buck Bunny (torrent file)
   3. Start torrent download
   4. Wait for the download to complete and look at the stats. Check that the download speed is zero (or not showing up at all) and not stuck at some non-zero number.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_webtorrent/extension/background/events/webtorrentEvents.ts [hunk 1]
```diff
@@ -3,9 +3,24 @@
...
 export const addWebtorrentEvents = (webtorrent: Instance) => {
   webtorrent.on('error', (e: Error | string) => {
     console.log('WebTorrent error: ', e)
   })
+
+  // Always update stats at least once every 5 seconds, even when no torrent
+  // events fire
+  interval = setInterval(() => {
+    webtorrent.torrents.forEach(torrent => {
+      // webtorrentActions.progressUpdated(torrent)
```
Comment (yrliou) [inline components/brave_webtorrent/extension/background/events/webtorrentEvents.ts]: same question as above.

File: components/brave_webtorrent/extension/background/events/webtorrentEvents.ts [hunk 2]
```diff
@@ -3,9 +3,24 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import { Instance } from 'webtorrent'
+// import webtorrentActions from '../actions/webtorrentActions'
```
Comment (yrliou) [inline components/brave_webtorrent/extension/background/events/webtorrentEvents.ts]: why is this commented out?

File: components/brave_webtorrent/extension/background/events/webtorrentEvents.ts (hunk 1 above)
Comment (feross) [reply components/brave_webtorrent/extension/background/events/webtorrentEvents.ts]: Good catch – sending too many PRs too quickly! Will fix

File: components/brave_webtorrent/extension/background/events/webtorrentEvents.ts (hunk 2 above)
Comment (feross) [reply components/brave_webtorrent/extension/background/events/webtorrentEvents.ts]: Will fix!

**Comment by feross** [comment]
Okay, uncommented the lines!

**Comment by feross** [comment]
@yrliou Running into a weird Travis error. My guess is that there's a cyclic dependency when I include the `import` that I initially had commented out. Lmk if you have any ideas.

**Comment by feross** [comment]
Figured it out. There's indeed some weird dependency issue. I found a workaround with `jest.mock` already in the test file. Just followed the model in that file to also mock `webtorrentEvents` and now it's working!

**Comment by feross** [comment]
@yrliou Is this safe to merge, despite the build failure?

**Comment by yrliou** [comment]
@feross CI failed at init stage which we can't have much information on other stages cause they're not executed.
Although I think the risk of this change is quite low, it would be great to have CI to at least build the PR, could you rebase this PR to kick a new PR build?
Thanks!

**Comment by yrliou** [comment]
Lost connection to win-64 CI node when building, but other platforms are passing.
This PR should be safe enough to merge.

**Pull request merged**